.github/workflows/compile.yml

```
@@ -17,9 +17,6 @@ jobs:
       - name: Check out repository code
         uses: actions/checkout@v4
 
-      - name: Ensure pre-commit checks pass
```

This only takes a few seconds, do we really need to split it?

Also how about just adding another entry under jobs instead of a completely new file?

---

I think splitting it into a separate job is advantageous for things like #867. Agreed that a separate job in the same file would work well though.

---

I really hate precommit check stops my hope of seeing the result of build. This make me angry.

---

We could also think about enabling pre-commit using the pre-commit ci app if we're changing this anyway: https://pre-commit.ci.

---

I'd suggest running `pre-commit install` locally, but I can see why you'd want to see the build result. However I don't think this should be a seaprate build file, if anything we want a matrix build and a condition.

It's unfortunate the githubs `continue-on-error` doesn't quite seem to do the right thing, I believe it will mark the whole job as a success rather than just continuing and marking it as failure.

I'm not sure I'd want the application that automatically applies the fixes since then we have to teach everyone how to pull the latest changes.

---

They can run it locally but put it in a seprate file makes it easy to disable it for fork users

---

I still think we don't want this in a separate file. Just install pre-commit locally. You can avoid the mess of `pip` now too:

```
curl -LsSf https://astral.sh/uv/install.sh | sh
uv tool install pre-commit
cd sail-riscv
pre-commit [--hook-type pre-push]
```